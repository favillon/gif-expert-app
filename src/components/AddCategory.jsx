import  PropTypes from 'prop-types'
import { useState } from 'react'

/*
    Forma de hacer el cambio del estado desde el hijo
    Key #1
    export const AddCategory = ({setCategories}) => {
*/
export const AddCategory = ({onNewCategory}) => {
  const [inputValue, setinputValue] = useState('One Punch')
  const onInputChange = ({target}) => {
    setinputValue(target.value)
  }
  const onSubmit = (event) => {
    event.preventDefault()
    if (inputValue.trim().length<=1) return;
    /*
        Forma de hacer el cambio del estado desde el hijo
        Key #1
        setCategories( categories => [inputValue, ...categories])
    */
    onNewCategory(inputValue.trim())
    setinputValue('')
  }
  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
       type="text"
       placeholder="Buscar gif"
       value={inputValue}
       onChange={onInputChange}
      />
    </form>
  )
}
AddCategory.propTypes = {
  onNewCategory : PropTypes.func.isRequired
}