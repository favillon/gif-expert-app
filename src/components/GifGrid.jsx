
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem';

export const GifGrid = ({category}) => {
  const {images, isLoading} = useFetchGifs(category)
  
  console.log({images, isLoading});

  return (
    <div>
      <h3>{category}</h3>
      {
        isLoading && (<h3>Cargando...</h3>)
      }

      <div className="card-grid">
        {
          images.map((image) => (
            <GifItem key={image.id} { ...image }/>
          ))
        }
      </div>
    </div>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
