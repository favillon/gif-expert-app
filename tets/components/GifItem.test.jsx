import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Test GifItem.jsx', () => {
    const title = 'Mi titulo';
    const url = 'https://favillon.xyz';
    test('Match snapshot', () => {
        const {container}  = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot()
    })


    test('should imagen y el alt De la imagen', () => {
        render(<GifItem title={title} url={url} />)
        // screen.debug() // ## Depurar el codigo generado
        expect(screen.getByRole('img').src).toContain(url)
        expect(screen.getByRole('img').alt).toBe(title)
        const {src, alt} = screen.getByRole('img')
        expect(src).toContain(url)
        expect(alt).toBe(title)
    })

    test('should mostrar el titulo', () => {
        render(<GifItem title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy()
    })
})