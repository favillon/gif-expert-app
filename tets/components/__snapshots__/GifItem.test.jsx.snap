// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifItem.jsx Match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Mi titulo"
      src="https://favillon.xyz"
    />
    Mi titulo
  </div>
</div>
`;
